src/powershell/Public/Remove-FinOpsCostExport.ps1

```diff
@@ -0,0 +1,169 @@
+    .PARAMETER RemoveData
+        Optional. Indicates that all cost data associated with the Export scope should be deleted.
+        This will delete all data in the storage account associated with the export scope (billing, subscription, management group, resource group).
+
+    .PARAMETER APIVersion
+        Optional. Specify an alternative API version. Default is 2023-04-01-preview.
+
+    .EXAMPLE
+        Remove-FinOpsCostExport -Name MyExport -Scope "/subscriptions/00000000-0000-0000-0000-000000000000" -RemoveData
+
+        Deletes a Cost Management Export named MyExport scoped to /subscriptions/00000000-0000-0000-0000-000000000000, and deletes all data associated with that scope.
+#>
+
+function Remove-FinOpsCostExport
+{
+  [CmdletBinding(SupportsShouldProcess)]
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Name,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Scope,
+
+    [Parameter()]
+    [switch]
+    $RemoveData,
+
+    [Parameter()]
+    [string]
+    $APIVersion = '2023-04-01-preview'
+  )
+
+  try
+  {
+    $additionalParamaters = @{
+      scope = $scope
+    }
+    $payload = ConvertTo-Json -InputObject $additionalParamaters -Depth 100
+
+    # TODO : Validate parameters and resources before proceeding
+    $httpResponse = Invoke-AzRestMethod `
+      -ResourceProviderName "Microsoft.CostManagement" `
+      -ResourceType "exports" `
+      -Name $Name `
+      -ApiVersion $ApiVersion `
+      -Method "GET" `
+      -Payload $payload
+
+    if ($httpResponse.StatusCode -eq 200)
+    {
+      # Export deleted successfully
+    }
+    elseif ($httpResponse.StatusCode -eq 404)
+    {
+      # Not found
+    }
+    else
+    {
+      # Error response describing why the operation failed.
+      throw "Not Exists: Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+    }
+
+    if ($PSCmdlet.ShouldProcess($Name, 'DeleteCostExport'))
+    {
+
+      # Using the REST API to delete the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "DELETE" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+        # Export deleted successfully
+      }
+      elseif ($httpResponse.StatusCode -eq 404)
+      {
+        # Not found - nothing to delete
+      }
+      else
+      {
+        # Error response describing why the operation failed.
+        throw "Delete Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+      }
+    }
+
+    # Delete associated ingestion data from storage account
+    if ($RemoveData)
+    {
+      # Using the REST API to get the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
```

Comment:
since we use this multiple times, we should define these parameters once in a hashtable then splat them whenever we need them: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting?view=powershell-7.3

Reply:
Might also be worth creating a helper for Cost Management REST calls, given we'll use that a lot.

Reply:
Done, although the splat override doesn't work for me...